src/main/java/uk/gov/hmcts/reform/iacaseapi/domain/handlers/presubmit/UpdateTribunalDecisionHandler.java

```diff
@@ -94,49 +93,42 @@ public PreSubmitCallbackResponse<AsylumCase> handle(PreSubmitCallbackStage callb
                 newDecisionAndReasons.setDocumentAndReasonsDocument(correctedDecisionAndReasonsDoc);
                 newDecisionAndReasons.setDateDocumentAndReasonsDocumentUploaded(dateProvider.now().toString());
                 newDecisionAndReasons.setSummariseChanges(summariseChanges);
-            }
 
-            Optional<List<IdValue<DecisionAndReasons>>> maybeExistingDecisionAndReasons =
-                asylumCase.read(CORRECTED_DECISION_AND_REASONS);
-            List<IdValue<DecisionAndReasons>> allCorrectedDecisions =
-                decisionAndReasonsAppender.append(newDecisionAndReasons, maybeExistingDecisionAndReasons.orElse(emptyList()));
+                Optional<List<IdValue<DecisionAndReasons>>> maybeExistingDecisionAndReasons =
+                    asylumCase.read(CORRECTED_DECISION_AND_REASONS);
+                List<IdValue<DecisionAndReasons>> allCorrectedDecisions =
+                    decisionAndReasonsAppender.append(newDecisionAndReasons, maybeExistingDecisionAndReasons.orElse(emptyList()));
 
-            asylumCase.write(CORRECTED_DECISION_AND_REASONS, allCorrectedDecisions);
+                asylumCase.write(CORRECTED_DECISION_AND_REASONS, allCorrectedDecisions);
+                asylumCase.write(UPDATE_TRIBUNAL_DECISION_DATE, dateProvider.now().toString());
```

Comment:
We need to write to UPDATE_TRIBUNAL_DECISION_DATE regardless of If it is YES or NO for value UPDATE_TRIBUNAL_DECISION_AND_REASONS_FINAL_CHECK. (Where it was originally)